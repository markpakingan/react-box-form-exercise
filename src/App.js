// App.js file

import './App.css';
import BoxList from './Boxlist';
import React from 'react';
import TodoList from './TodoList';
function App() {



  return (
    <div className="App">
     

     {/* <BoxList/> */}

    <TodoList />


    </div>
  );
}

export default App;
