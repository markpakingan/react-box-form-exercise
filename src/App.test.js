import React from "react";
import App from "./App";
import { render, fireEvent} from "@testing-library/react";
import BoxList from "./Boxlist";
import NewToDoForm from "./NewTodoForm";



it("renders without crashing", () =>{
  render(<App/>)
});


// snapshot for App.js
it("matches snapshot for App.js", ()=>{
  const {asFragment} = render(<App/>)
  expect(asFragment()).toMatchSnapshot();
})

// snapshot for BoxList.js
it("matches snapshot for boxlist", ()=>{
  const {asFragment} = render(<BoxList/>)
  expect(asFragment()).toMatchSnapshot();
})

test("input field updates state correctly", () => {
  const { getByPlaceholderText } = render(<NewToDoForm addTask={() => {}} />);

  const input = getByPlaceholderText("task");

  fireEvent.change(input, { target: { value: "Task 1" } });

  expect(input.value).toBe("Task 1");
});

