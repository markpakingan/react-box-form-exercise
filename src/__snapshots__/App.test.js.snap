// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for App.js 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div>
      <form>
        <label
          for="backgroundcolor"
        >
          Background Color: 
        </label>
        <input
          name="backgroundcolor"
          placeholder="backgroundcolor"
          type="text"
          value=""
        />
        <label
          for="width"
        >
          Width: 
        </label>
        <input
          name="width"
          placeholder="width"
          type="text"
          value=""
        />
        <label
          for="height"
        >
          Height : 
        </label>
        <input
          name="height"
          placeholder="height"
          type="text"
          value=""
        />
        <button>
          Submit!
        </button>
      </form>
    </div>
    <div>
      <div>
        <label
          for="task"
        >
          Things To Do: 
        </label>
        <input
          name="task"
          placeholder="task"
          type="text"
          value=""
        />
        <button>
          AddTask
        </button>
      </div>
      <ul />
    </div>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot for boxlist 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="backgroundcolor"
      >
        Background Color: 
      </label>
      <input
        name="backgroundcolor"
        placeholder="backgroundcolor"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width: 
      </label>
      <input
        name="width"
        placeholder="width"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height : 
      </label>
      <input
        name="height"
        placeholder="height"
        type="text"
        value=""
      />
      <button>
        Submit!
      </button>
    </form>
  </div>
</DocumentFragment>
`;
